src/frontend/components/LTINav/index.spec.tsx

```diff
@@ -0,0 +1,212 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import { wrapInIntlProvider } from '../../utils/tests/intl';
+import { wrapInRouter } from '../../utils/tests/router';
+import React from 'react';
+import { LTINav } from './index';
+import {
+  documentMockFactory,
+  videoMockFactory,
+} from '../../utils/tests/factories';
+import { uploadState, Video } from '../../types/tracks';
+import { Document } from '../../types/file';
+import { Route, Switch } from 'react-router-dom';
+import { PLAYER_ROUTE } from '../routes';
+import { DASHBOARD_ROUTE } from '../Dashboard/route';
+import { modelName } from '../../types/models';
+
+let mockModelName: any;
+let mockCanUpdate: boolean;
+let mockMaintenance: boolean;
+jest.mock('../../data/appData', () => ({
+  appData: {
+    get modelName() {
+      return mockModelName;
+    },
+  },
+  getDecodedJwt: () => ({
+    permissions: {
+      can_update: mockCanUpdate,
+    },
+    maintenance: mockMaintenance,
+  }),
+}));
+
+describe('<LTINav />', () => {
+  const testLTINav = (object: Video | Document) =>
+    render(
+      wrapInIntlProvider(
+        wrapInRouter(
+          <React.Fragment>
+            <LTINav object={object} />
+            <Switch>
+              <Route
+                exact
+                path={PLAYER_ROUTE()}
+                render={({ match }) => <div>playerView</div>}
+              />
+              <Route
+                exact
+                path={DASHBOARD_ROUTE()}
+                render={({ match }) => <div>dashboardView</div>}
+              />
+            </Switch>
```

Comment:
Can you explain why there's a need to use another component to test yours? You can't do it another way to just test your component on its own?
Does the switch need to be added for all tests?

Reply:
Yes, you're right, there's a better way.
You can check it there https://reactrouter.com/web/guides/testing/checking-location-in-tests